bundles/org.openhab.binding.atlona/src/main/java/org/openhab/binding/atlona/internal/pro3/AtlonaPro3PortocolHandler.java

```diff
@@ -280,6 +289,94 @@ String login() throws Exception {
         return "Password was invalid - please check your atlona setup";
     }
 
+    /**
+     * Attempts to log into the older HD model switches using a slightly different protocol
+     *
+     * @return a null if logged in successfully (or if switch didn't require login). Non-null if an exception occurred.
+     * @throws IOException an IO exception occurred during login
+     */
+    String loginHD() throws Exception {
```

Comment:
```suggestion
    String loginHD() throws IOException {
```

Reply:
This whole method looks very much the same as `loginHD`. Do you think it is possible to have a common `login(boolean isUHD)` whis is called from `loginHD` and `loginUHD` and takes care of the differences?

Reply:
I tried to do it in one method but the logic was complicated. I do not have the other model to test with so I decided to make a separate method to avoid breaking the core functionality of the binding.

Reply:
ok